oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -192,6 +201,61 @@ String createTokenUrlWithScopes() {
     return tokenUrl.toString();
   }
 
+  /**
+   * Gets the universe domain from the GCE metadata server.
+   *
+   * <p>Returns an explicit universe domain if it was provided during credential initialization.
+   *
+   * <p>Returns the {@link Credentials#GOOGLE_DEFAULT_UNIVERSE} if universe domain endpoint is
+   * unavailable or returns an empty string.
+   *
+   * <p>Otherwise, returns universe domain from GCE metadata service.
+   *
+   * <p>Any above value is cached for the credential lifetime.
```

Comment:
qq, What is the default credential lifetime? Say if the lifetime is like 1 hour (as an example), would that mean that a subsequent `getUniverseDomain()` call in an hour would make another network call since the universe domain value is no longer cached?

Reply:
There are two key entities: credential and token

Credential lifetime is the object lifetime. Effectively unlimited. 
Token lifetime is 1 hour. 

The universe domain is a property of the credential. We fetch it once and cache for object's lifetime. Every token for the credential expected to target that universe.

Reply:
Understood, thanks! nit: Perhaps update to `The Universe Domain value is cached...`?

Reply:
Agree )